Source/WTF/wtf/generic/MainThreadGeneric.cpp

```diff
@@ -32,32 +32,22 @@
 
 #include "config.h"
 #include <pthread.h>
-#if HAVE(PTHREAD_NP_H)
-#include <pthread_np.h>
-#endif
 
 #include <wtf/RunLoop.h>
 
 namespace WTF {
 
-#if !HAVE(PTHREAD_MAIN_NP)
 static pthread_t mainThread;
-#endif
 
 void initializeMainThreadPlatform()
 {
-#if !HAVE(PTHREAD_MAIN_NP)
+    // mainThread is often but not always the initial process thread
```

Comment:
We have a code style rule that comments are sentences and should end in periods:

```suggestion
    // mainThread is often but not always the initial process thread.
```

But I would go even farther here to describe things a bit more:

```suggestion
    // WebKit APIs must be consistently used from exactly one thread. The thread that initializes WebKit
    // is considered the "WebKit main thread," and it is an error to use WebKit APIs from any other thread.
    // The WebKit main thread need not be the application's actual OS-level main thread, which might be
    // controlled by a language runtime or virtual machine; for example, in Eclipse, the OS main thread is
    // controlled by the JVM, while the separate WebKit main therad runs all the GUI and WebKit stuff.
```

This is different from Apple systems, but that's fine.